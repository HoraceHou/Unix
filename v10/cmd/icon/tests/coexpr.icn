record array(a,b,c,d,e,f,g)
procedure main()
   write(image(x := [array(),table(),write,main,[],&input,1,"abc",'aa',2.0,&null,create 1]) | "failed")
   write(image(x := sort(x)) | "failed")
   write(image(every write(image(!x))) | "failed")
   write(image(e := create 1 to 10) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(e := ^e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(@e) | "failed")
   write(image(*e) | "failed")
   write(image(*e) | "failed")
   write(image(image(e)) | "failed")
   write(image(tab := create {write("entering tab"); 1 | 2 | 3}) | "failed")
   write(image(trim := create {write("entering trim"); @tab | (main @:= tab)}) | "failed")
   write(image(@trim) | "failed")
   write(image(@trim) | "failed")
   write(image(write(image(trim))) | "failed")
   write(image(write(image(tab))) | "failed")
   write(image(write(image(main))) | "failed")
end
