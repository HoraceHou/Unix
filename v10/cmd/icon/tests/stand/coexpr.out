list(12)
list(12)
&null
1
2.0
"abc"
'a'
co-expression(0)
&input
function write
procedure main
list(0)
table(0)
record array(7)
failed
co-expression(0)
1
2
3
co-expression(0)
1
2
3
4
5
6
7
8
9
10
failed
failed
failed
failed
10
10
"co-expression(10)"
co-expression(0)
co-expression(0)
entering trim
entering tab
1
2
co-expression(2)
"co-expression(2)"
co-expression(2)
"co-expression(2)"
2
"2"
